### PR TITLE
Untangling: fix focus style on panel sidebar

### PR DESCRIPTION
**Pull request by fushar**
Related to https://github.com/Automattic/wp-calypso/pull/95927#pullrequestreview-2406985882

## Proposed Changes

This PR removes the unnecessary margin (see linked comment above), by fixing the hover/focus styling of the sidebar items.

I "invented" the hover background color. We can refine the color later.

For the keyboard focus style, I copied from here: 

https://github.com/Automattic/wp-calypso/blob/fc74e34e0216373e7a75c72075d1437d16e5b0d1/client/components/section-nav/item.scss#L70-L72

### With mouse

https://github.com/user-attachments/assets/41686f2c-89a6-481d-9faa-fd0618b9fb71

### With Tab key (keyboard)

https://github.com/user-attachments/assets/cbf49a7b-c00c-44c8-a8a7-14954b811d74

## Why are these changes being made?

We want to get rid of excessive margin.

## Testing Instructions

1. Go to /sites
2. Click any site
3. Click Settings tab
4. Loop through the subtabs using mouse and keyboard as explained above.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/sites/components/panel-sidebar/style.scss [hunk 1]
```diff
@@ -18,20 +18,33 @@
 	list-style: none;
 }
 
-.panel-sidebar-tab {
+.components-button.panel-sidebar-tab {
 	border-radius: 4px;
 	width: 100%;
-	margin: 2px 0;
```
Comment (lsl) [inline client/sites/components/panel-sidebar/style.scss:24]: Maybe:
```
margin: 0 0 2px 0;
```
Without it, mousing over the item below looks kind of blocky. I think we want a little bit of space:

Screenshot_2024-11-01_11-52-08
Screenshot_2024-11-01_11-51-47

**Review by lsl — APPROVED** [review]
minor margin thing, but lgtm

File: client/sites/components/panel-sidebar/style.scss (hunk 1 above)
Comment (fushar) [reply client/sites/components/panel-sidebar/style.scss:24]: Oops, I missed that the Figma do actually contain the margin 🤦‍♂️ added it!

**Review by arthur791004 — APPROVED** [review]
Tested and looks good to me!

**Pull request merged**